spec/requests/surveys_controllers_spec.rb

```diff
@@ -9,4 +9,26 @@
       expect(response).to have_http_status(:ok)
     end
   end
+
+  describe 'qrcode' do
+    let(:survey) { create :survey }
+
+    context 'when there is no administrator signed in' do
+      it 'redirects to surveys list' do
+        get qrcode_survey_path(survey)
+        expect(response).to redirect_to(surveys_url(locale: I18n.locale))
```

Comment:
I don't think that we should redirect to the surveys if you're not signed in. I'd prefer rendering a 404 page or redirect to the login page with the sign in notice, like when you access `/admin` directly without being signed in.